advanced/dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx

```diff
@@ -548,8 +554,9 @@ export function JsonRpcContextProvider({
             `Missing rpcProvider definition for chainId: ${chainId}`
           );
         }
+        if(lastTxId == undefined) throw new Error(`Last transaction id  ${lastTxId}, make sure call to sendCalls returns successfully. `);
```

Comment:
Better to use `===` strict comparison here if the possible values are `string | undefined`

```suggestion
        if (lastTxId === undefined) throw new Error(`Last transaction ID is undefined, make sure previous call to sendCalls returns successfully. `);
```